### PR TITLE
Fixes #7454: Directives with double quotes in their name need to be escaped before going in rudder_directives.cf and rudder_expected_reports.csv

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/7454

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala [hunk 1]
```diff
@@ -86,6 +86,76 @@ trait PrepareTemplateVariables {
...
+  /*
+   * utilitary method for formating an input list
+   */
+  def formatBundleFileInputs(x: Seq[String]) = {
+    val inputs = x.distinct
+    if (inputs.isEmpty) {
+      ""
+    } else {
+      inputs.mkString("\"", s"""",\n${" "*4}"""", s"""",""")
+    }
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala]: In fact you don't need the spaces

current output (with spaces)is: 

    inputs => {
      "rpmPackageInstallation/7.0/rpmPackageInstallation.cf",
          "test_3_2/1.0/test_3_2.cf",
    };

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala [hunk 2]
```diff
@@ -86,6 +86,76 @@ trait PrepareTemplateVariables {
...
+   * Sort the techniques according to the order of the associated BundleOrder of Cf3PolicyDraft.
+   * Sort at best: sort rule then directives, and take techniques on that order, only one time
+   * Sort system directive first.
+   *
+   * CAREFUL: this method only take care of sorting based on "BundleOrder", other sorting (like
+   * "system must go first") are not taken into account here !
+   */
+  def sortTechniques(nodeId: NodeId, techniques: Seq[Technique], container: Cf3PolicyDraftContainer): Seq[(Technique, List[BundleOrder])] = {
+
+    def sortByOrder(tech: Seq[Technique], container: Cf3PolicyDraftContainer): Seq[(Technique, List[BundleOrder])] = {
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala]: Why using this function ? you call it directly afterwars without treatment before :(

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala (hunk 1 above)
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala]: Mybe you'll need to adapt the test :)

**Comment by fanf** [comment]
PR updated

**Pull request merged**